src/api/queues/v0/queues.ts

```diff
@@ -104,80 +104,35 @@ export class Queue {
    *     value: "test"
    *   };
    * });
-   * ```
-   */
-  send = async (tasks: Task | Task[]): Promise<void | FailedMessage[]> => {
-    if (Array.isArray(tasks)) {
-      return this.sendBatch(tasks);
-    }
-
-    return new Promise((resolve, reject) => {
-      const request = new QueueSendRequest();
-
-      request.setTask(taskToWire(tasks));
-      request.setQueue(this.name);
-
-      this.queueing.QueueServiceClient.send(request, (error) => {
-        if (error) {
-          reject(fromGrpcError(error));
-        } else {
-          resolve();
-        }
-      });
-    });
-  };
-
-  /**
-   * Send a collection of tasks to a queue, which can be retrieved by other services.
-   *
-   * @param tasks the tasks to push to the queue
-   * @returns a list containing details of any tasks that failed to publish.
-   *
-   * Example:
-   * ```typescript
-   * import { Queueing } from "@nitric/sdk"
-   *
-   * const queueing = new Queueing();
-   *
-   * const failedTasks = await queueing.queue("my-queue").sendBatch([{
-   *   payloadType: "my-payload";
-   *   payload: {
-   *     value: "test"
-   *   };
-   * }]);
-   *
-   * // do something with failedTasks
-   * // console.log(failedTasks);
-   * ```
    */
-  private sendBatch = async (tasks: Task[]): Promise<FailedMessage[]> => {
+   async send(tasks: Task): Promise<FailedMessage>;
+   async send(tasks: Task[]): Promise<FailedMessage[]>;
+   async send(tasks: Task | Task[]): Promise<FailedMessage | FailedMessage[]> {
```

Comment:
Just chatting to Tim, we were thinking the single task implementation should probably have a `void` response (and throw an exception if the send fails). That will provide more consistency with other methods like publishing events.

Reply:
just did some quick reading apparently `never` might be more conventional in the case where we throw an error:
```typescript
async send(tasks: Task): Promise<never>;
```

Reply:
makes sense, go with `void`.